TLM/TMPE.API/Manager/IJunctionRestrictionsManager.cs

```diff
@@ -574,5 +576,17 @@ public interface IJunctionRestrictionsManager {
         /// Updates the default values for all junction restrictions and segments.
         /// </summary>
         void UpdateAllDefaults();
+
+        event Action<FlagsChangedEventArgs> FlagsChanged;
```

Comment:
what is this?

Reply:
It's incomplete. It's an event that will be invoked when junction restrictions flags change on a segment end, but `FlagsChangedEventArgs` is missing some properties right now.

Great example of this being a work in progress. I've just created the draft pull request so there's visibility on what I'm doing. 🙂

Reply:
To further elaborate, it's intended to be a clear and supported way for mods like AN to know when junction restrictions change. That way you don't have to detect changes based on assumptions about the implementation. Instead, you have a direct promise that this will always work.

Reply:
> mods like AN to know when junction restrictions change. 


isn't that what notifier does?

Reply:
> > mods like AN to know when junction restrictions change.
> 
> isn't that what notifier does?

Yes, but `INotifier` is limited in the amount of specific information it can share about the change. By adding specific events at the manager level, you can provide more useful information. And if implemented correctly, when no one is using them all they cost is a check for null.

Reply:
> although the event args is just an object and do not have hard-coded type safety. 

And that's the key difference. Manager-owned events put less responsibility on the consumer. Fewer opportunities for error.